### ISSUE TITLE
Quickcheck output isn't (entirely) reproducible

### ISSUE THREAD

**Issue by kbara** [Igalia/pflua#77]
```
./pflua-quickcheck savefiles/wingolog.org.pcap
Using time as seed: 668464616
--- Expanded:
{ "if",
  { "!=",
    { "int32",
      { "ntohl",
        { "int32",
          { "ntohs",
            { "ntohl",
              "len" } } } } },
    0 },
  { "<",
    { "int32",
      { "ntohs",
        { "/",
          1,
          { "int32",
            { "ntohl",
              { "int32",
                { "ntohs",
                  { "ntohl",
                    "len" } } } } } } } },
    "len" },
  { "fail" } }
--- Perturbed:
{ "if",
  { "!=",
    { "int32",
      { "ntohl",
        { "int32",
          { "ntohs",
            { "ntohl",
              "len" } } } } },
    0 },
  { "<",
    { "int32",
      { "ntohs",
        { "/",
          1,
          { "int32",
            { "ntohl",
              { "int32",
                { "ntohs",
                  { "ntohl",
                    "len" } } } } } } } },
    "len" },
  { "fail" } }
--- Optimized:
{ "if",
  { "!=",
    { "ntohl",
      "len" },
    0 },
  { "<",
    { "ntohs",
      { "/",
        1,
        { "ntohl",
          { "ntohs",
            { "ntohl",
              "len" } } } } },
    "len" },
  { "fail" } }
function: 0x4073e608
function: 0x4073ff18
4228
Optimized and unoptimized filter return values do not match.
Run again as: pflua-quickcheck --seed=668464616 --iterations=175 savefiles/wingolog.org.pcap 
make: *** [quickcheck] Error 1
```

-----
Running the given command line led to the same output 3 times; on the 4th, it diverged:
```
./pflua-quickcheck --seed=668464616 --iterations=175 savefiles/wingolog.org.pcap
175 iterations succeeded.
```
Similar things happen without --iterations; sometimes it generates the above test case, and sometimes it doesn't:
```
./pflua-quickcheck --seed=668464616  savefiles/wingolog.org.pcap
1000 iterations succeeded.
```

And occasionally, it's reproduced, but with a different number of iterations. The following both have the same optimized code as above, but yield:
Run again as: pflua-quickcheck --seed=668464616 --iterations=166 savefiles/wingolog.org.pcap 
Run again as: pflua-quickcheck --seed=668464616 --iterations=180 savefiles/wingolog.org.pcap 

Tests should be entirely reproducible with the seed.

**Comment by andywingo** [comment]
https://gist.github.com/andywingo/d1800650a72e1062c3ee is a standalone test case.

**Comment by andywingo** [comment]
Well!  I got a small reproducible test case for this, linked in the above gist, but it seems that the bug was fixed in somewhere between 66515a054c0826cee4f0abc5e532f35b421e9c81 and 7f013005f61b82300d4ec591fd4cec59a74d62ff of luajit -- so a luajit bug!  I'll update our luajit copy in pflua.

**Comment by andywingo** [comment]
Fixed in #115 , woo!
